sql/plan/insert.go

```diff
@@ -129,7 +160,19 @@ func (p *InsertInto) WithChildren(children ...sql.Node) (sql.Node, error) {
 		return nil, sql.ErrInvalidChildrenNumber.New(p, len(children), 2)
 	}
 
-	return NewInsertInto(children[0], children[1], p.Columns), nil
+	return NewInsertInto(children[0], children[1], p.Columns, p.Ignore), nil
+}
+
+func (p *InsertInto) ValidateValueCount(ctx *sql.Context) error {
+	switch node := p.Right.(type) {
+	case *Values:
+		for _, exprTuple := range node.ExpressionTuples {
+			if len(exprTuple) != len(p.Columns) {
+				return ErrInsertIntoMismatchValueCount.New()
+			}
+		}
+	}
+	return nil
```

Comment:
Better to return an error here than let an unsupported p.Right give bogus results.